src/platform.ts

```diff
@@ -55,196 +60,437 @@ export function getPlatformDetails(): IPlatformDetails {
 
-/**
- * Gets a list of all available PowerShell instance on the specified platform.
- * @param platformDetails Specifies information about the platform - primarily the operating system.
- * @param sessionSettings Specifies the user/workspace settings. Additional PowerShell exe paths loaded from settings.
- * @returns An array of IPowerShellExeDetails objects with the PowerShell name & exe path for each instance found.
- */
-export function getAvailablePowerShellExes(
-    platformDetails: IPlatformDetails,
-    sessionSettings: Settings.ISettings | undefined): IPowerShellExeDetails[] {
-
-    let paths: IPowerShellExeDetails[] = [];
-
-    if (platformDetails.operatingSystem === OperatingSystem.Windows) {
-        if (platformDetails.isProcess64Bit) {
-            paths.push({
-                versionName: WindowsPowerShell64BitLabel,
-                exePath: System32PowerShellPath,
-            });
-
-            paths.push({
-                versionName: WindowsPowerShell32BitLabel,
-                exePath: SysWow64PowerShellPath,
-            });
+        // Find the base directory for MSIX application exe shortcuts
+        const msixAppDir = path.join(process.env.LOCALAPPDATA, "Microsoft", "WindowsApps");
+
+        if (!fs.existsSync(msixAppDir)) {
+            return null;
+        }
+
+        // Define whether we're looking for the preview or the stable
+        let pwshMsixDirRegex: RegExp = null;
+        let pwshMsixName: string = null;
+        if (options && options.findPreview) {
+            pwshMsixDirRegex = PowerShellExeFinder.PwshPreviewMsixRegex;
+            pwshMsixName = "PowerShell Preview MSIX";
         } else {
-            if (platformDetails.isOS64Bit) {
-                paths.push({
-                    versionName: WindowsPowerShell64BitLabel,
-                    exePath: SysnativePowerShellPath,
-                });
+            pwshMsixDirRegex = PowerShellExeFinder.PwshMsixRegex;
+            pwshMsixName = "PowerShell MSIX";
+        }
+
+        // We should find only one such application, so return on the first one
+        for (const subdir of fs.readdirSync(msixAppDir)) {
+            if (pwshMsixDirRegex.test(subdir)) {
+                const pwshMsixPath = path.join(msixAppDir, subdir, "pwsh.exe");
+                return new PossiblePowerShellExe(pwshMsixPath, pwshMsixName);
+            }
+        }
+
+        // If we find nothing, return null
+        return null;
+    }
+
+    private findPSCoreStableSnap(): IPossiblePowerShellExe {
+        return new PossiblePowerShellExe(SnapExePath, "PowerShell Snap");
+    }
+
+    private findPSCorePreviewSnap(): IPossiblePowerShellExe {
+        return new PossiblePowerShellExe(SnapPreviewExePath, "PowerShell Preview Snap");
+    }
+
+    private findPSCoreWindowsInstallation(options?: {
+        findNonNativeBitness?: boolean; findPreview?: boolean }): IPossiblePowerShellExe {
+
+        const findPreview = options && options.findPreview || false;
```

Comment:
```suggestion
        const findPreview = !!(options && options.findPreview);
```
😈 (don't actually do this lol)